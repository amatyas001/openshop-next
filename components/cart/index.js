export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
