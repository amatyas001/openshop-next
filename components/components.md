Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
