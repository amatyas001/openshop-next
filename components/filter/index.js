export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
