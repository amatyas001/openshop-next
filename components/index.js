export * from './shared';
export * from './cart';
export * from './payment';
