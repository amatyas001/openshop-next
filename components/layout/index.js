export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
