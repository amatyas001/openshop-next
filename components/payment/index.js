export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
