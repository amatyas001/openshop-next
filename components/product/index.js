export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
