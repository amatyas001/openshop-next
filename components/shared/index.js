export * from './Button/Button';
export * from './Card/Card';
export * from './Field/Field';
export * from './Rating/Rating';
export * from './ScrollTop/ScrollTop';
export * from './Navbar/Navbar';
export * from './Footer/Footer';
export * from './Spinner/Spinner';
