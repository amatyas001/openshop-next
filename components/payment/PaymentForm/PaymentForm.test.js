import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as actions from '@app/redux/thunk/intent';
import { paymentReview } from '@app/redux/actions';
import { PaymentForm } from '@app/components';

const mockStore = configureStore([thunk]);

const completeForm = (form) => {
  let details = {};

  form.root.findAllByProps({ 'data-testid': `input` }).map((item) => {
    details[item.props.id] = 'mock_value';
    act(() => {
      item.props.onChange({ target: { value: 'mock_value' } });
    });
  });

  return details;
};

describe('<PaymentForm />', () => {
  let tree, store, submitButton, backButton;

  beforeAll(() => {
    store = mockStore({
      payment: { status: 'form', token: 'token' },
      cart: [{ id: 'id' }],
      amount: 100,
    });

    act(() => {
      tree = create(
        <Provider store={store}>
          <PaymentForm />
        </Provider>
      );
    });

    submitButton = tree.root.findByProps({
      'data-testid': 'form-submit-button',
    });

    backButton = tree.root.findByProps({
      'data-testid': 'form-review-button',
    });
  });

  it('should render without props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render disabled button', () => {
    expect(submitButton.props.disabled).toBeTruthy();
  });

  it('should push to review state', () => {
    act(() => {
      backButton.props.onClick();
    });
    expect(store.getActions()).toEqual([paymentReview()]);
  });

  describe('if form complete', () => {
    let details;

    beforeAll(() => {
      details = completeForm(tree);
    });

    it('should render enabled button', () => {
      expect(submitButton.props.disabled).toBeFalsy();
    });

    it('should send intent with details', () => {
      const intent = jest
        .spyOn(actions, 'paymentIntent')
        .mockImplementation(() => jest.fn());

      act(() => {
        submitButton.props.onClick();
      });

      expect(intent).toHaveBeenCalledTimes(1);
      expect(intent).toHaveBeenCalledWith(
        details,
        store.getState().cart,
        store.getState().payment.token
      );
    });
  });
});
