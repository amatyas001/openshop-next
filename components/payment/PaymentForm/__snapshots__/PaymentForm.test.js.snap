// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentForm /> should render without props 1`] = `
<form
  className="css-8o331q"
>
  <fieldset
    className="css-94sf3x"
  >
    <fieldset
      className="css-9x3bhi"
    >
      <label
        className="css-1xnp38i"
        htmlFor="name"
      >
        name
      </label>
      <input
        className="css-cth1sm"
        id="name"
        onChange={[Function]}
        placeholder="John Doe"
        required={true}
        type="text"
        value=""
      />
    </fieldset>
    <fieldset
      className="css-9x3bhi"
    >
      <label
        className="css-1xnp38i"
        htmlFor="address"
      >
        address
      </label>
      <input
        className="css-cth1sm"
        id="address"
        onChange={[Function]}
        placeholder="J191 S. Illinois Street, Everett, MA 02149"
        required={true}
        type="text"
        value=""
      />
    </fieldset>
    <fieldset
      className="css-9x3bhi"
    >
      <label
        className="css-1xnp38i"
        htmlFor="email"
      >
        email
      </label>
      <input
        className="css-cth1sm"
        id="email"
        onChange={[Function]}
        placeholder="johndoe@example.com"
        required={true}
        type="email"
        value=""
      />
    </fieldset>
    <fieldset
      className="css-9x3bhi"
    >
      <label
        className="css-1xnp38i"
        htmlFor="phone"
      >
        phone
      </label>
      <input
        className="css-cth1sm"
        id="phone"
        onChange={[Function]}
        placeholder="(941) 555-0123"
        required={true}
        type="tel"
        value=""
      />
    </fieldset>
  </fieldset>
  <p
    className="css-1yab5hb"
  >
    By proceeding you accept the
     
    <a
      className="css-1c5zfe9"
    >
      Terms of Service
    </a>
     
    of the company.
  </p>
  <div
    className="css-1rsrz83"
  >
    <button
      className="css-1h13z07"
      data-cy="button"
      data-testid="form-review-button"
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="button"
    >
      BACK TO REVIEW
    </button>
    <button
      className="css-187kagv"
      data-cy="button"
      data-testid="form-submit-button"
      disabled={true}
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      FILL THE FORM
    </button>
  </div>
</form>
`;
